include/fiction/algorithms/simulation/sidb/critical_temperature.hpp

```diff
@@ -241,14 +229,28 @@ class critical_temperature_impl
             // is used to provide 100 % accuracy for the Critical Temperature).
             simulation_results = quickexact(layout, qe_params);
         }
-        else
+#if (FICTION_ALGLIB_ENABLED)
+        else if (params.engine == sidb_simulation_engine::CLUSTERCOMPLETE)
+        {
+            const clustercomplete_params<cell<Lyt>> cc_params{params.simulation_parameters};
+
+            // All physically valid charge configurations are determined for the given layout (`ClusterComplete`
+            // simulation is used to provide 100 % accuracy for the Critical Temperature).
+            simulation_results = clustercomplete(layout, cc_params);
+        }
+#endif  // FICTION_ALGLIB_ENABLED
```

Comment:
Shall we add an `#else` case with an `assert` or something notifying the user that `ALGLIB` must be installed if *ClusterComplete* is to be used?

This would then apply to all such cases.